data_processing/pathology/cli/dsa/dsa_viz.py

```diff
@@ -39,49 +41,49 @@ def save_push_results(base_annotation, elements, annotation_name, image_filename
 
     dsa_annotation["name"] = annotation_name
 
-    dsa_uuid = get_item_uuid(image_filename, uri, token)
+    image_id = re.search(image_id_regex, image_filename).group(1)
+    annotation_name_replaced = annotation_name.replace(" ","_")
 
-    if not dsa_uuid:
-        print("ERROR: could not find item in DSA matching image name:" , image_filename)
-        return
+    outfile_name = os.path.join(output_folder, f"{annotation_name_replaced}_{image_id}.json")
 
-    push_annotation_to_dsa_image(dsa_uuid, dsa_annotation, uri, token)
+    try:
+        with open(outfile_name, 'w') as outfile:
+            json.dump(dsa_annotation, outfile)
+        return outfile_name
+    except Exception as e:
+        print("ERROR: write permissions needs to be enabled for: ", os.path.dirname(outfile_name))
+        return None
 
 
-@click.group()
-@click.pass_context
-@click.option('-c', '--config',
-              help="json including DSA host, port and token info",
-              type=click.Path(exists=True),
-              required=True)
-def cli(ctx, config):
-    """
-    DSA visualization CLI
-    """
-    ctx.ensure_object(dict)
-
-    with open(config) as config_json:
-        config_data = json.load(config_json)
-
-    # Girder Token can be found in the DSA API Swagger Docs under 'token': (http://{host}:8080/api/v1#!/token/token_currentSession)
-    ctx.obj["uri"] = config_data["host"] + ":" + config_data["port"]
-    ctx.obj["token"] = config_data["token"]
-    # print(ctx.obj)
-
-    # TODO use girder client
-    # https://girder.readthedocs.io/en/latest/python-client.html#the-python-client-library
-
-    # check DSA connection
-    system_check(ctx.obj["uri"], ctx.obj["token"])
-
-
-@cli.command()
-@click.pass_context
+@click.command()
 @click.option("-d", "--data_config",
               help="path to data config file",
               required=True,
               type=click.Path(exists=True))
-def stardist_polygon(ctx, data_config):
+@click.option("-s", "--source_type",
+              help="string describing data source that is to be transformed into a dsa json. \
+               supports stardist-polygon, stardist-cell, regional-polygon, qupath-polygon, bitmask-polygon, heatmap",
+              required=True)
+def cli(data_config, source_type):
+    """
+    DSA annotation builder
+    """
+
+    functions = {"stardist-polygon": stardist_polygon, 
+                "stardist-cell": stardist_cell, 
+                "regional-polygon": regional_polygon, 
+                "qupath-polygon": qupath_polygon, 
+                "bitmask-polygon": bitmask_polygon,
+                "heatmap": heatmap}
+    
+    if source_type not in functions:
+            print(f"ERROR: source_type {source_type} is not supported")
+    else:   
+        annotation_filepath = functions[source_type](data_config)
```

Comment:
Nice

Reply:
Haha looked up some pythonic alternatives to if/else..